gubernator/view_build.py

```diff
@@ -194,18 +206,26 @@ def get(self, prefix, job, build):
             return
         started, finished, results = details
 
-        pr, pr_path, repo = parse_pr_path(prefix)
-        pr_digest = None
-        if pr:
-            pr_digest = models.GHIssueDigest.get(repo, pr)
-
+        want_build_log = False
         build_log = ''
         build_log_src = None
         if 'log' in self.request.params or (not finished) or \
             (finished and finished.get('result') != 'SUCCESS' and len(results['failed']) <= 1):
+            want_build_log = True
             build_log = get_build_log(build_dir)
-            if not build_log:
-                build_log, build_log_src = get_running_build_log(job, build, repo)
+
+        pr, pr_path, pr_digest, repo = None, None, None, None
+        for item in self.app.config["external_services"].values():
+            if prefix.startswith(item["gcs_pull_prefix"]):
+                pr, pr_path, repo = parse_pr_path(
+                    gcs_path=prefix,
+                    default_org=self.app.config['default_org'],
+                    default_repo=self.app.config['default_repo'],
+                )
+                pr_digest = models.GHIssueDigest.get(repo, pr)
+                if want_build_log and not build_log:
```

Comment:
Had to rejigger this logic a bit. Technically it's no longer identical behavior as before, but before we would pass in the `repo` which was `None` if this was not a PR job. Then we'd try to look up the Prow URL for a `None` and default to the k8s Prow. This would mean non-PR jobs in progress for non-k8s repos would get the k8s link (a broken link). Now we just don't put a link down at all.